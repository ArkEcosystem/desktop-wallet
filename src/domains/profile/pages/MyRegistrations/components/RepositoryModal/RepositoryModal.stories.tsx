import { boolean, withKnobs } from "@storybook/addon-knobs";
import React from "react";

import { RepositoryModal } from "./RepositoryModal";

export default {
	title: "Domains / Profile / Pages / MyRegistrations / Components / RepositoryModal",
	decorators: [withKnobs],
};

const repositories = [
	{
		provider: "GitHub",
		url: "http://gitlab.com/robank",
	},
	{
		provider: "GitLab",
		url: "http://gitlab.com/robank",
	},
	{
		provider: "BitBucket",
		url: "http://bitbucket.com/robank",
	},
	{
		provider: "Npm",
		url: "http://npmjs.com/robank",
	},
];

export const Default = () => (
	<RepositoryModal
		isOpen={boolean("Is Open", true)}
		handleClose={() => alert("closed")}
		repositories={repositories}
	/>
);
