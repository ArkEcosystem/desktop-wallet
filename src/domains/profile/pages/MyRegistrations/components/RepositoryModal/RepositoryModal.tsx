import { Modal } from "app/components/Modal";
import React from "react";
import { useTranslation } from "react-i18next";

import { RepositoryLink, RepositoryLinkProps } from "./components/RepositoryLink";

type Props = {
	isOpen?: boolean;
	handleClose?: any;
	repositories?: any;
};

export const RepositoryModal = ({ isOpen, handleClose, repositories }: Props) => {
	const { t } = useTranslation();

	return (
		<Modal isOpen={isOpen} onClose={handleClose} title={t("PROFILE.MODAL_REPOSITORIES.TITLE")} size="sm">
			<div className="flex flex-col mt-5 space-y-5">
				{repositories.map(({ provider, url }: RepositoryLinkProps) => (
					<RepositoryLink key={provider} provider={provider} url={url} />
				))}
			</div>
		</Modal>
	);
};

RepositoryModal.defaultProps = {
	isOpen: false,
	repositories: [],
};
