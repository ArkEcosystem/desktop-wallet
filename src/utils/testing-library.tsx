import { ARK } from "@arkecosystem/platform-sdk-ark";
import { Environment } from "@arkecosystem/platform-sdk-profiles";
import { render } from "@testing-library/react";
import { EnvironmentProvider } from "app/contexts";
import { i18n } from "app/i18n";
import { httpClient } from "app/services";
import { createMemoryHistory } from "history";
import React from "react";
import { I18nextProvider } from "react-i18next";
import { Router } from "react-router-dom";
import { StubStorage } from "tests/mocks";

import envFixture from "../tests/fixtures/env/data.json";

export const env = new Environment({ coins: { ARK }, httpClient, storage: new StubStorage(envFixture) });

const WithProviders: React.FC = ({ children }: { children?: React.ReactNode }) => (
	<I18nextProvider i18n={i18n}>
		<EnvironmentProvider env={env}>{children}</EnvironmentProvider>
	</I18nextProvider>
);

const customRender = (component: React.ReactElement, options: any = {}) =>
	render(component, { wrapper: WithProviders, ...options });

const renderWithRouter = (
	component: React.ReactElement,
	{ routes = ["/"], history = createMemoryHistory({ initialEntries: routes }), withProviders = true } = {},
) => {
	const RouterWrapper = ({ children }: { children: React.ReactNode }) =>
		withProviders ? (
			<WithProviders>
				<Router history={history}>{children}</Router>
			</WithProviders>
		) : (
			<Router history={history}>{children}</Router>
		);

	return {
		...customRender(component, { wrapper: RouterWrapper }),
		history,
	};
};

export * from "@testing-library/react";

export { customRender as render, renderWithRouter };
