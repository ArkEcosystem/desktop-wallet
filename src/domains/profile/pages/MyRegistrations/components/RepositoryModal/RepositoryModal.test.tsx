import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "testing-library";

import { RepositoryModal } from "./RepositoryModal";

describe("RepositoryModal", () => {
	const repositories = [
		{
			provider: "GitHub",
			url: "http://github.com/robank",
		},
		{
			provider: "GitLab",
			url: "http://gitlab.com/robank",
		},
		{
			provider: "BitBucket",
			url: "http://bitbucket.com/robank",
		},
		{
			provider: "Npm",
			url: "http://npmjs.com/robank",
		},
	];

	it("should render empty state", () => {
		const handleClose = jest.fn();

		const { asFragment } = render(
			<Router>
				<RepositoryModal isOpen repositories={repositories} handleClose={() => handleClose()} />,
			</Router>,
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
