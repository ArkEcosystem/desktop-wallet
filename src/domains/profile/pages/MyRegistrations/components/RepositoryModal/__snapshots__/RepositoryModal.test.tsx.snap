// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryModal should render empty state 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 z-50 bg-black opacity-50"
    data-testid="modal__overlay"
  />
  <div
    class="sc-AxiKw wLhVq fixed top-0 left-0 right-0 z-50 flex flex-col px-10 pt-6 pb-8 mx-auto mt-24 overflow-hidden rounded-xl bg-theme-background"
    data-testid="modal__inner"
  >
    <div
      class="absolute top-0 right-0 z-50 mt-5 mr-5"
    >
      <button
        class="sc-AxjAm bThAoP"
        color="neutral"
        data-testid="modal__close-btn"
        type="button"
      >
        <div
          class="p-1"
        >
          <div
            class="sc-AxirZ hDwaov"
            height="12"
            width="12"
          >
            <svg>
              cross-slim.svg
            </svg>
          </div>
        </div>
      </button>
    </div>
    <div
      class="py-4"
    >
      <h2
        class="mb-0 text-3xl font-bold"
      >
        Repository
      </h2>
      <div
        class="flex-1"
      >
        <div
          class="flex flex-col mt-5 space-y-5"
        >
          <div
            class="flex items-center space-x-5"
          >
            <div
              class="sc-AxhCb bzWLYJ border-theme-neutral-800"
            >
              <div
                class="sc-AxirZ hmuEJr"
                data-testid="repository-link__icon"
                height="22"
                width="22"
              >
                <svg>
                  github.svg
                </svg>
              </div>
            </div>
            <div
              class="flex flex-col space-y-1"
            >
              <span
                class="text-sm text-theme-neutral-400"
              >
                GitHub
              </span>
              <a
                class="text-theme-primary"
                href="http://github.com/robank"
                target="_blank"
              >
                http://github.com/robank
              </a>
            </div>
          </div>
          <div
            class="flex items-center space-x-5"
          >
            <div
              class="sc-AxhCb bzWLYJ border-theme-neutral-800"
            >
              <div
                class="sc-AxirZ hmuEJr"
                data-testid="repository-link__icon"
                height="22"
                width="22"
              >
                <svg>
                  gitlab.svg
                </svg>
              </div>
            </div>
            <div
              class="flex flex-col space-y-1"
            >
              <span
                class="text-sm text-theme-neutral-400"
              >
                GitLab
              </span>
              <a
                class="text-theme-primary"
                href="http://gitlab.com/robank"
                target="_blank"
              >
                http://gitlab.com/robank
              </a>
            </div>
          </div>
          <div
            class="flex items-center space-x-5"
          >
            <div
              class="sc-AxhCb bzWLYJ border-theme-neutral-800"
            >
              <div
                class="sc-AxirZ hmuEJr"
                data-testid="repository-link__icon"
                height="22"
                width="22"
              >
                <svg>
                  bitbucket.svg
                </svg>
              </div>
            </div>
            <div
              class="flex flex-col space-y-1"
            >
              <span
                class="text-sm text-theme-neutral-400"
              >
                Bitbucket
              </span>
              <a
                class="text-theme-primary"
                href="http://bitbucket.com/robank"
                target="_blank"
              >
                http://bitbucket.com/robank
              </a>
            </div>
          </div>
          <div
            class="flex items-center space-x-5"
          >
            <div
              class="sc-AxhCb bzWLYJ border-theme-neutral-800"
            >
              <div
                class="sc-AxirZ hmuEJr"
                data-testid="repository-link__icon"
                height="22"
                width="22"
              >
                <svg>
                  npm.svg
                </svg>
              </div>
            </div>
            <div
              class="flex flex-col space-y-1"
            >
              <span
                class="text-sm text-theme-neutral-400"
              >
                NPM
              </span>
              <a
                class="text-theme-primary"
                href="http://npmjs.com/robank"
                target="_blank"
              >
                http://npmjs.com/robank
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
